walk-in-interview/src/main/java/com/google/job/data/Job.java

```diff
@@ -1,18 +1,146 @@
-    public Job() {}
+    public Job() {
+        this.jobId = "";
+        this.jobStatus = null;
+        this.jobTitle = "";
+        this.jobLocation = null;
+        this.jobDescription = "";
+        this.jobPay = null;
+        this.requirements = null;
+        this.postExpiry = 0;
+        this.jobDuration = null;
+    }
+
+    public static JobBuilder newBuilder() {
+        return new JobBuilder();
+    }
+
+    public static final class JobBuilder {
+        // Optional parameters - initialized to default values
+        private String jobId = "";
+
+        // TODO(issue/25): merge the account stuff into job post.
+
+        private JobStatus jobStatus;
+        private String jobTitle;
+        private Location location;
+        private String jobDescription;
+        private JobPayment jobPay;
+        private List<String> requirements;
+        private long postExpiry; // a timestamp
+        private JobDuration jobDuration;
+
+        private JobBuilder() {}
+
+        // TODO(issue/19): add more validation check
+
+        public JobBuilder setJobId(String jobId) {
+            this.jobId = jobId;
+            return this;
+        }
+
+        public JobBuilder setJobStatus(JobStatus jobStatus) {
+            this.jobStatus = jobStatus;
+            return this;
+        }
+
+        public JobBuilder setJobTitle(String jobTitle) throws IllegalArgumentException {
+            if (jobTitle.isEmpty()) {
+                throw new IllegalArgumentException("Empty job title provided");
+            }
+
+            this.jobTitle = jobTitle;
+            return this;
+        }
+
+        public JobBuilder setLocation(Location location) {
+            this.location = location;
+            return this;
+        }
 
-    public Job(String jobTitle) throws IllegalArgumentException {
-        if (jobTitle.isEmpty()) {
-            throw new IllegalArgumentException("Empty job title provided");
+        public JobBuilder setJobDescription(String jobDescription) throws IllegalArgumentException {
+            if (jobDescription.isEmpty()) {
+                throw new IllegalArgumentException("Empty job description provided");
+            }
+
+            this.jobDescription = jobDescription;
+            return this;
+        }
+
+        public JobBuilder setJobPay(JobPayment jobPay) {
+            this.jobPay = jobPay;
+            return this;
+        }
+
+        public JobBuilder setRequirements(List<String> requirements) {
+            this.requirements = ImmutableList.copyOf(requirements);
+            return this;
+        }
+
+        public JobBuilder setPostExpiry(long postExpiry) {
+            this.postExpiry = postExpiry;
+            return this;
+        }
+
+        public JobBuilder setJobDuration(JobDuration jobDuration) {
+            this.jobDuration = jobDuration;
+            return this;
+        }
+
+        public Job build() {
+            return new Job(this);
```

Comment:
you need validation of your arguments:
- check for arguments that were not provided, nullable is not a valid value.

Reply:
Fixed